{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Runs with env: Python [conda env: allele_specific]**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Steps:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "  Steps:  \n",
    "  \n",
    "    Bash:        \n",
    "        1) Map and split with SNPsplit using sc_allele_align_split.sh script\n",
    "        2) Generate VCF using sc_allele_generate_vcf.sh script\n",
    "    Python:  \n",
    "        1) Import annotated vcf file (scikit-allele) as Pandas DataFrame  \n",
    "        2) Clean the dataframe  \n",
    "        3) Filter for the number of SNPs/gene and number of reads/SNP  \n",
    "        4) Construct allelic count matrix (export_2)  \n",
    "        5) Filter the matrix for min n reads per sum of the two alleles  \n",
    "        6) Create matrix with allelic ratios (export_3)  \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import io\n",
    "import os\n",
    "import re\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Timepoint</th>\n",
       "      <th>Xi</th>\n",
       "      <th>Cell</th>\n",
       "      <th>GC_index1_index2</th>\n",
       "      <th>Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Day_9_Xi_Mus_33</td>\n",
       "      <td>Day_9</td>\n",
       "      <td>Mus</td>\n",
       "      <td>33</td>\n",
       "      <td>GC071392_ACTCTAGG-AAGAGGCA</td>\n",
       "      <td>Day_9_Xi_Mus_33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Day_10_Xi_Mus_25</td>\n",
       "      <td>Day_10</td>\n",
       "      <td>Mus</td>\n",
       "      <td>25</td>\n",
       "      <td>GC071392_ACTCTAGG-ACTCGCTA</td>\n",
       "      <td>Day_10_Xi_Mus_25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Day_10_Xi_Mus_32</td>\n",
       "      <td>Day_10</td>\n",
       "      <td>Mus</td>\n",
       "      <td>32</td>\n",
       "      <td>GC071392_ACTCTAGG-ACTGAGCG</td>\n",
       "      <td>Day_10_Xi_Mus_32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Day_9_Xi_Mus_27</td>\n",
       "      <td>Day_9</td>\n",
       "      <td>Mus</td>\n",
       "      <td>27</td>\n",
       "      <td>GC071392_ACTCTAGG-AGGCAGAA</td>\n",
       "      <td>Day_9_Xi_Mus_27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Day_9_Xi_Mus_36</td>\n",
       "      <td>Day_9</td>\n",
       "      <td>Mus</td>\n",
       "      <td>36</td>\n",
       "      <td>GC071392_ACTCTAGG-ATCTCAGG</td>\n",
       "      <td>Day_9_Xi_Mus_36</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Unnamed: 0 Timepoint   Xi  Cell            GC_index1_index2  \\\n",
       "0   Day_9_Xi_Mus_33     Day_9  Mus    33  GC071392_ACTCTAGG-AAGAGGCA   \n",
       "1  Day_10_Xi_Mus_25    Day_10  Mus    25  GC071392_ACTCTAGG-ACTCGCTA   \n",
       "2  Day_10_Xi_Mus_32    Day_10  Mus    32  GC071392_ACTCTAGG-ACTGAGCG   \n",
       "3   Day_9_Xi_Mus_27     Day_9  Mus    27  GC071392_ACTCTAGG-AGGCAGAA   \n",
       "4   Day_9_Xi_Mus_36     Day_9  Mus    36  GC071392_ACTCTAGG-ATCTCAGG   \n",
       "\n",
       "               Name  \n",
       "0   Day_9_Xi_Mus_33  \n",
       "1  Day_10_Xi_Mus_25  \n",
       "2  Day_10_Xi_Mus_32  \n",
       "3   Day_9_Xi_Mus_27  \n",
       "4   Day_9_Xi_Mus_36  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation = pd.read_csv(\"/staging/leuven/stg_00041/Adrian/TALON_JANISZEWSKI_XCR2/allele_specific/pre_processing/AJ_0065_annotation.csv\")\n",
    "annotation.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create dictionary from index name and Name\n",
    "\n",
    "index_name_dict = dict(zip(annotation.GC_index1_index2, annotation.Name))\n",
    "#dict(list(index_name_dict.items())[:3])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Borrowed from internet\n",
    "def vcfRead(path):\n",
    "    with open(path, 'r') as f:\n",
    "        lines = [l for l in f if not l.startswith('##')]\n",
    "    return pd.read_csv(\n",
    "        io.StringIO(''.join(lines)),\n",
    "        dtype={'#CHROM': str, 'POS': int, 'ID': str, 'REF': str, 'ALT': str,\n",
    "               'QUAL': str, 'FILTER': str, 'INFO': str},\n",
    "        sep='\\t'\n",
    "    ).rename(columns={'#CHROM': 'CHROM'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = vcfRead(\"/staging/leuven/stg_00041/Adrian/TALON_JANISZEWSKI_XCR2/allele_specific/pre_processing/allelic_merged_annotated.vcf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>output/sorted_bam/GC071392_ACTCTAGG-AAGAGGCA_Aligned.out.genome1.sorted.bam</th>\n",
       "      <th>...</th>\n",
       "      <th>output/sorted_bam/GC071398_TCTTACGC-GGAGCTAC_Aligned.out.genome1.sorted.bam</th>\n",
       "      <th>output/sorted_bam/GC071398_TCTTACGC-GGAGCTAC_Aligned.out.genome2.sorted.bam</th>\n",
       "      <th>output/sorted_bam/GC071398_TCTTACGC-TACGCTGC_Aligned.out.genome1.sorted.bam</th>\n",
       "      <th>output/sorted_bam/GC071398_TCTTACGC-TACGCTGC_Aligned.out.genome2.sorted.bam</th>\n",
       "      <th>output/sorted_bam/GC071398_TCTTACGC-TAGCGCTC_Aligned.out.genome1.sorted.bam</th>\n",
       "      <th>output/sorted_bam/GC071398_TCTTACGC-TAGCGCTC_Aligned.out.genome2.sorted.bam</th>\n",
       "      <th>output/sorted_bam/GC071398_TCTTACGC-TCGACGTC_Aligned.out.genome1.sorted.bam</th>\n",
       "      <th>output/sorted_bam/GC071398_TCTTACGC-TCGACGTC_Aligned.out.genome2.sorted.bam</th>\n",
       "      <th>output/sorted_bam/GC071398_TCTTACGC-TGCAGCTA_Aligned.out.genome1.sorted.bam</th>\n",
       "      <th>output/sorted_bam/GC071398_TCTTACGC-TGCAGCTA_Aligned.out.genome2.sorted.bam</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>3205860</td>\n",
       "      <td>.</td>\n",
       "      <td>N</td>\n",
       "      <td>C,&lt;*&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>DP=3</td>\n",
       "      <td>DP</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>3205889</td>\n",
       "      <td>.</td>\n",
       "      <td>N</td>\n",
       "      <td>G,&lt;*&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>DP=3</td>\n",
       "      <td>DP</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>3206270</td>\n",
       "      <td>.</td>\n",
       "      <td>N</td>\n",
       "      <td>T,&lt;*&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>DP=2;GENE=\"Xkr4\"</td>\n",
       "      <td>DP</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>3274462</td>\n",
       "      <td>.</td>\n",
       "      <td>N</td>\n",
       "      <td>C,&lt;*&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>DP=4;GENE=\"Xkr4\"</td>\n",
       "      <td>DP</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>3301254</td>\n",
       "      <td>.</td>\n",
       "      <td>N</td>\n",
       "      <td>C,&lt;*&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>DP=2;GENE=\"Xkr4\"</td>\n",
       "      <td>DP</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1333 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS ID REF    ALT QUAL FILTER              INFO FORMAT  \\\n",
       "0     1  3205860  .   N  C,<*>    0   PASS              DP=3     DP   \n",
       "1     1  3205889  .   N  G,<*>    0   PASS              DP=3     DP   \n",
       "2     1  3206270  .   N  T,<*>    0   PASS  DP=2;GENE=\"Xkr4\"     DP   \n",
       "3     1  3274462  .   N  C,<*>    0   PASS  DP=4;GENE=\"Xkr4\"     DP   \n",
       "4     1  3301254  .   N  C,<*>    0   PASS  DP=2;GENE=\"Xkr4\"     DP   \n",
       "\n",
       "  output/sorted_bam/GC071392_ACTCTAGG-AAGAGGCA_Aligned.out.genome1.sorted.bam  \\\n",
       "0                                                  .                            \n",
       "1                                                  .                            \n",
       "2                                                  .                            \n",
       "3                                                  .                            \n",
       "4                                                  .                            \n",
       "\n",
       "   ...  \\\n",
       "0  ...   \n",
       "1  ...   \n",
       "2  ...   \n",
       "3  ...   \n",
       "4  ...   \n",
       "\n",
       "  output/sorted_bam/GC071398_TCTTACGC-GGAGCTAC_Aligned.out.genome1.sorted.bam  \\\n",
       "0                                                  .                            \n",
       "1                                                  .                            \n",
       "2                                                  .                            \n",
       "3                                                  .                            \n",
       "4                                                  .                            \n",
       "\n",
       "  output/sorted_bam/GC071398_TCTTACGC-GGAGCTAC_Aligned.out.genome2.sorted.bam  \\\n",
       "0                                                  .                            \n",
       "1                                                  .                            \n",
       "2                                                  .                            \n",
       "3                                                  .                            \n",
       "4                                                  .                            \n",
       "\n",
       "  output/sorted_bam/GC071398_TCTTACGC-TACGCTGC_Aligned.out.genome1.sorted.bam  \\\n",
       "0                                                  .                            \n",
       "1                                                  .                            \n",
       "2                                                  .                            \n",
       "3                                                  .                            \n",
       "4                                                  .                            \n",
       "\n",
       "  output/sorted_bam/GC071398_TCTTACGC-TACGCTGC_Aligned.out.genome2.sorted.bam  \\\n",
       "0                                                  .                            \n",
       "1                                                  .                            \n",
       "2                                                  .                            \n",
       "3                                                  .                            \n",
       "4                                                  .                            \n",
       "\n",
       "  output/sorted_bam/GC071398_TCTTACGC-TAGCGCTC_Aligned.out.genome1.sorted.bam  \\\n",
       "0                                                  .                            \n",
       "1                                                  .                            \n",
       "2                                                  .                            \n",
       "3                                                  .                            \n",
       "4                                                  .                            \n",
       "\n",
       "  output/sorted_bam/GC071398_TCTTACGC-TAGCGCTC_Aligned.out.genome2.sorted.bam  \\\n",
       "0                                                  .                            \n",
       "1                                                  .                            \n",
       "2                                                  .                            \n",
       "3                                                  .                            \n",
       "4                                                  .                            \n",
       "\n",
       "  output/sorted_bam/GC071398_TCTTACGC-TCGACGTC_Aligned.out.genome1.sorted.bam  \\\n",
       "0                                                  .                            \n",
       "1                                                  .                            \n",
       "2                                                  .                            \n",
       "3                                                  .                            \n",
       "4                                                  .                            \n",
       "\n",
       "  output/sorted_bam/GC071398_TCTTACGC-TCGACGTC_Aligned.out.genome2.sorted.bam  \\\n",
       "0                                                  .                            \n",
       "1                                                  .                            \n",
       "2                                                  .                            \n",
       "3                                                  .                            \n",
       "4                                                  .                            \n",
       "\n",
       "  output/sorted_bam/GC071398_TCTTACGC-TGCAGCTA_Aligned.out.genome1.sorted.bam  \\\n",
       "0                                                  .                            \n",
       "1                                                  .                            \n",
       "2                                                  .                            \n",
       "3                                                  .                            \n",
       "4                                                  .                            \n",
       "\n",
       "  output/sorted_bam/GC071398_TCTTACGC-TGCAGCTA_Aligned.out.genome2.sorted.bam  \n",
       "0                                                  .                           \n",
       "1                                                  .                           \n",
       "2                                                  .                           \n",
       "3                                                  .                           \n",
       "4                                                  .                           \n",
       "\n",
       "[5 rows x 1333 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def vcfClean(vcf_df: pd.DataFrame, \n",
    "             renameSamples: bool = True, \n",
    "             renameDict: dict = None): \n",
    "    \n",
    "    \"\"\"\n",
    "    Takes vcf as dataframe imported by vcfRead function, cleans and renames the columns.\n",
    "    \n",
    "    Parameters:\n",
    "    arg1 (dataframe): Output of vcfRead.\n",
    "    \n",
    "    arg2 (boolean): Specificy if samples should be renamed against the dictionary. Default: True\n",
    "    \n",
    "    arg3 (dict {chr:chr}): Dictionary containing current name : new name pairs. If arg2 is True.\n",
    "\n",
    "    Returns:\n",
    "    DataFrame:Returns DataFrame with CHROM, ...samples..., DP (total reads across samples), GENE as columns, each row is a single SNP. Compatible with vcfFilter.\n",
    "   \"\"\"\n",
    "\n",
    "    assert isinstance(vcf_df, pd.DataFrame)\n",
    "    \n",
    "    # Split INFO field to DP and GENE\n",
    "    \n",
    "    vcf_df['INFO'] = vcf_df['INFO'].str.replace('|'.join(map(re.escape, ['GENE=','DP', '\"', \"=\"])), '') # Clean DP and GENE columns\n",
    "    \n",
    "    vcf_clean = vcf_df.join(df['INFO'].str.split(';', 1, expand=True).rename(columns={0:'DP', 1:'GENE'})) # Separate to DP and GENE columns\n",
    "    \n",
    "    # Keep only SNPs annotated with GENE\n",
    "    vcf_clean = vcf_clean[vcf_clean['GENE'].notnull()]\n",
    "               \n",
    "    vcf_clean.drop([\"POS\", \"ID\", \"REF\",\"ALT\", \"QUAL\", \"FILTER\", \"INFO\", \"FORMAT\"], axis=1, inplace = True) \n",
    "    \n",
    "\n",
    "    # Clean colnames\n",
    "        \n",
    "    vcf_clean.columns = vcf_clean.columns.str.replace('|'.join(map(re.escape, ['output/sorted_bam/', '_Aligned.out', \".sorted.bam\"])), '') \n",
    "    vcf_clean.columns = vcf_clean.columns.str.replace('.genome1','.129').str.replace('.genome2','.Cast')\n",
    "    \n",
    "    # Rename samples against provided dictionary\n",
    "    \n",
    "    assert isinstance(renameSamples, bool)\n",
    "    assert isinstance(renameDict, dict)\n",
    "    \n",
    "    if renameSamples: \n",
    "        for index, name in renameDict.items():\n",
    "            vcf_clean.columns = vcf_clean.columns.str.replace(index,name)\n",
    "    \n",
    "     \n",
    "    return vcf_clean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def vcfFilter (df: pd.DataFrame,\n",
    "              minSNPcov: int = 2,\n",
    "              minSNPGene: int = 4):\n",
    "    \"\"\"\n",
    "    Takes output of vcfClean function, filters the SNPs based on the read coverage across all cells, filters the Genes based on the number of SNPs.\n",
    "    \n",
    "    Parameters:\n",
    "    arg1 (dataframe): Output of vcfClean.\n",
    "    \n",
    "    arg2 (int): Minimum number of reads each SNPs has to be covered with across all cells. Default = 2\n",
    "    \n",
    "    arg3 (int): Minimum number of SNPs each gene has to cover. Default = 2\n",
    "\n",
    "    Returns:\n",
    "    DataFrame:Returns DataFrame with CHROM, ...samples..., GENE as columns, each row is a single SNP. \n",
    "   \"\"\"\n",
    "    # Filter SNPs based on read coverage\n",
    "    # Filter out the Genes that have less than 2 SNPs\n",
    "    \n",
    "    assert isinstance(minSNPcov, int)\n",
    "    assert isinstance(minSNPGene, int)\n",
    "    \n",
    "    df[\"DP\"] = df[\"DP\"].astype(float) # For safety from NAs keep floats\n",
    "    \n",
    "    df_filtered = df[df['DP'] >= minSNPcov].groupby(\"GENE\").filter(lambda x: len(x) >= minSNPGene)\n",
    "    \n",
    "    df_filtered.drop([\"DP\"], axis=1, inplace = True) \n",
    "    \n",
    "    return df_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = vcfClean(df, renameDict = index_name_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = vcfFilter(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>Day_9_Xi_Mus_33.129</th>\n",
       "      <th>Day_9_Xi_Mus_33.Cast</th>\n",
       "      <th>Day_10_Xi_Mus_25.129</th>\n",
       "      <th>Day_10_Xi_Mus_25.Cast</th>\n",
       "      <th>Day_10_Xi_Mus_32.129</th>\n",
       "      <th>Day_10_Xi_Mus_32.Cast</th>\n",
       "      <th>Day_9_Xi_Mus_27.129</th>\n",
       "      <th>Day_9_Xi_Mus_27.Cast</th>\n",
       "      <th>Day_9_Xi_Mus_36.129</th>\n",
       "      <th>...</th>\n",
       "      <th>Day_12_Xi_Cast_38.Cast</th>\n",
       "      <th>Day_12_Xi_Cast_41.129</th>\n",
       "      <th>Day_12_Xi_Cast_41.Cast</th>\n",
       "      <th>Day_12_Xi_Cast_43.129</th>\n",
       "      <th>Day_12_Xi_Cast_43.Cast</th>\n",
       "      <th>Day_12_Xi_Cast_48.129</th>\n",
       "      <th>Day_12_Xi_Cast_48.Cast</th>\n",
       "      <th>Day_12_Xi_Cast_47.129</th>\n",
       "      <th>Day_12_Xi_Cast_47.Cast</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Xkr4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Xkr4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Xkr4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Xkr4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Xkr4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1326 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM Day_9_Xi_Mus_33.129 Day_9_Xi_Mus_33.Cast Day_10_Xi_Mus_25.129  \\\n",
       "2     1                   .                    .                    .   \n",
       "3     1                   .                    .                    .   \n",
       "4     1                   .                    .                    .   \n",
       "5     1                   .                    .                    .   \n",
       "6     1                   .                    .                    .   \n",
       "\n",
       "  Day_10_Xi_Mus_25.Cast Day_10_Xi_Mus_32.129 Day_10_Xi_Mus_32.Cast  \\\n",
       "2                     .                    .                     .   \n",
       "3                     .                    .                     .   \n",
       "4                     .                    .                     .   \n",
       "5                     .                    .                     .   \n",
       "6                     .                    .                     .   \n",
       "\n",
       "  Day_9_Xi_Mus_27.129 Day_9_Xi_Mus_27.Cast Day_9_Xi_Mus_36.129  ...  \\\n",
       "2                   .                    .                   .  ...   \n",
       "3                   .                    .                   .  ...   \n",
       "4                   .                    .                   .  ...   \n",
       "5                   .                    .                   .  ...   \n",
       "6                   .                    .                   .  ...   \n",
       "\n",
       "  Day_12_Xi_Cast_38.Cast Day_12_Xi_Cast_41.129 Day_12_Xi_Cast_41.Cast  \\\n",
       "2                      .                     .                      .   \n",
       "3                      .                     .                      .   \n",
       "4                      .                     .                      .   \n",
       "5                      .                     .                      .   \n",
       "6                      .                     .                      .   \n",
       "\n",
       "  Day_12_Xi_Cast_43.129 Day_12_Xi_Cast_43.Cast Day_12_Xi_Cast_48.129  \\\n",
       "2                     .                      .                     .   \n",
       "3                     .                      .                     .   \n",
       "4                     .                      .                     .   \n",
       "5                     .                      .                     .   \n",
       "6                     .                      .                     .   \n",
       "\n",
       "  Day_12_Xi_Cast_48.Cast Day_12_Xi_Cast_47.129 Day_12_Xi_Cast_47.Cast  GENE  \n",
       "2                      .                     .                      .  Xkr4  \n",
       "3                      .                     .                      .  Xkr4  \n",
       "4                      .                     .                      .  Xkr4  \n",
       "5                      .                     .                      .  Xkr4  \n",
       "6                      .                     .                      .  Xkr4  \n",
       "\n",
       "[5 rows x 1326 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/vsc-hard-mounts/leuven-data/320/vsc32023/jupyter_notebooks/Talon_Janiszewski_XCR2'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"/staging/leuven/stg_00041/Adrian/TALON_JANISZEWSKI_XCR2/allele_specific/pre_processing/AJ_0065_vcf_clean.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def vcf2matrix (df: pd.DataFrame):\n",
    "    \"\"\"\"\n",
    "    Takes output of vcfFilter function, creates the gene/sample matrix from the vcf data frame. Sums all the reads across the genes.\n",
    "    \n",
    "    Parameters:\n",
    "    arg1 (dataframe): Output of vcfFilter.\n",
    "    \n",
    "    Returns:\n",
    "    DataFrame:Returns matrix genes x samples\n",
    "   \"\"\"\n",
    "\n",
    "    matrix = df.set_index(['CHROM','GENE']).replace('.','0').astype('int32').groupby(['CHROM','GENE']).sum()\n",
    "    matrix = matrix.reset_index()\n",
    "\n",
    "    return matrix    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>GENE</th>\n",
       "      <th>Day_9_Xi_Mus_33.129</th>\n",
       "      <th>Day_9_Xi_Mus_33.Cast</th>\n",
       "      <th>Day_10_Xi_Mus_25.129</th>\n",
       "      <th>Day_10_Xi_Mus_25.Cast</th>\n",
       "      <th>Day_10_Xi_Mus_32.129</th>\n",
       "      <th>Day_10_Xi_Mus_32.Cast</th>\n",
       "      <th>Day_9_Xi_Mus_27.129</th>\n",
       "      <th>Day_9_Xi_Mus_27.Cast</th>\n",
       "      <th>...</th>\n",
       "      <th>Day_12_Xi_Cast_38.129</th>\n",
       "      <th>Day_12_Xi_Cast_38.Cast</th>\n",
       "      <th>Day_12_Xi_Cast_41.129</th>\n",
       "      <th>Day_12_Xi_Cast_41.Cast</th>\n",
       "      <th>Day_12_Xi_Cast_43.129</th>\n",
       "      <th>Day_12_Xi_Cast_43.Cast</th>\n",
       "      <th>Day_12_Xi_Cast_48.129</th>\n",
       "      <th>Day_12_Xi_Cast_48.Cast</th>\n",
       "      <th>Day_12_Xi_Cast_47.129</th>\n",
       "      <th>Day_12_Xi_Cast_47.Cast</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1500015O10Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1700001G17Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1700016C15Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1700019A02Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1700019D03Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1326 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM           GENE  Day_9_Xi_Mus_33.129  Day_9_Xi_Mus_33.Cast  \\\n",
       "0     1  1500015O10Rik                    0                     0   \n",
       "1     1  1700001G17Rik                    0                     0   \n",
       "2     1  1700016C15Rik                    0                     0   \n",
       "3     1  1700019A02Rik                    0                     0   \n",
       "4     1  1700019D03Rik                    0                     0   \n",
       "\n",
       "   Day_10_Xi_Mus_25.129  Day_10_Xi_Mus_25.Cast  Day_10_Xi_Mus_32.129  \\\n",
       "0                     0                      0                     0   \n",
       "1                     0                      0                     0   \n",
       "2                     0                      0                     0   \n",
       "3                     0                      0                     0   \n",
       "4                    12                      0                     0   \n",
       "\n",
       "   Day_10_Xi_Mus_32.Cast  Day_9_Xi_Mus_27.129  Day_9_Xi_Mus_27.Cast  ...  \\\n",
       "0                      0                    0                     0  ...   \n",
       "1                      0                    0                     0  ...   \n",
       "2                      0                    0                     0  ...   \n",
       "3                      0                    0                     0  ...   \n",
       "4                      0                    0                     0  ...   \n",
       "\n",
       "   Day_12_Xi_Cast_38.129  Day_12_Xi_Cast_38.Cast  Day_12_Xi_Cast_41.129  \\\n",
       "0                      0                       0                      0   \n",
       "1                      0                       0                      0   \n",
       "2                      0                       0                      0   \n",
       "3                      0                       0                      0   \n",
       "4                      0                       0                      0   \n",
       "\n",
       "   Day_12_Xi_Cast_41.Cast  Day_12_Xi_Cast_43.129  Day_12_Xi_Cast_43.Cast  \\\n",
       "0                       0                      0                       0   \n",
       "1                       0                      0                       0   \n",
       "2                       0                      0                       0   \n",
       "3                       0                      0                       0   \n",
       "4                       5                      2                       0   \n",
       "\n",
       "   Day_12_Xi_Cast_48.129  Day_12_Xi_Cast_48.Cast  Day_12_Xi_Cast_47.129  \\\n",
       "0                      0                       0                      0   \n",
       "1                      0                       0                      0   \n",
       "2                      0                       0                      0   \n",
       "3                      0                       0                      0   \n",
       "4                     35                       0                     18   \n",
       "\n",
       "   Day_12_Xi_Cast_47.Cast  \n",
       "0                       0  \n",
       "1                       0  \n",
       "2                       0  \n",
       "3                       2  \n",
       "4                       0  \n",
       "\n",
       "[5 rows x 1326 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = vcf2matrix(df)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"/staging/leuven/stg_00041/Adrian/TALON_JANISZEWSKI_XCR2/allele_specific/pre_processing/AJ_0065.9_allelic_count_mtx.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def matrix2ratio (df: pd.DataFrame, \n",
    "                  minSum: int = 0):\n",
    "    \"\"\"\"\n",
    "    Takes output of vcf2matrix function, creates a long table and calculates allelic ratio for all samples\n",
    "    \n",
    "    Parameters:\n",
    "    arg1 (dataframe): Output of vcf2matrix.\n",
    "    \n",
    "    arg2 (int): Minimum number of summed reads from two alleles per each gene and sample\n",
    "    \n",
    "    Returns:\n",
    "    DataFrame:Returns long dataframe with columns: chrom, gene, sample, genome\n",
    "   \"\"\"\n",
    "\n",
    "\n",
    "    df = df.drop(['CHROM'], axis=1)\n",
    "    df = df.set_index('GENE')\n",
    "    \n",
    "    df_t = df.transpose().reset_index()\n",
    "    \n",
    "    # Convert from wide to long format\n",
    "    \n",
    "    df_long = df_t.melt(var_name = 'gene', value_name = 'expression', id_vars = 'index')\n",
    "    \n",
    "    # Split column \n",
    "    \n",
    "    df_long = df_long.join(df_long['index'].str.split('\\.', 1, expand=True).rename(columns={0:'Name', 1:'genome'})).drop('index', axis=1)\n",
    "    \n",
    "    # create 129 and Cast columns\n",
    "    \n",
    "    df_allelic = df_long.pivot_table(index = ['Name','gene'], columns = 'genome', values = 'expression').reset_index()\n",
    "    \n",
    "    # Filter for the sum of reads from both alleles\n",
    "    \n",
    "    df_allelic['sumReads'] = df_allelic['129']+df_allelic['Cast']\n",
    "    df_allelic = df_allelic[df_allelic['sumReads'] >= minSum]\n",
    "    \n",
    "    # Calculate allelic ratios\n",
    "    \n",
    "    df_allelic['ratioPercent'] = df_allelic['129']/(df_allelic['129']+df_allelic['Cast'])+0.0001\n",
    "    df_allelic['ratioLog'] = np.log2((df_allelic['129'])/(df_allelic['Cast'])+0.0001)\n",
    "\n",
    "    return df_allelic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>genome</th>\n",
       "      <th>Name</th>\n",
       "      <th>gene</th>\n",
       "      <th>129</th>\n",
       "      <th>Cast</th>\n",
       "      <th>sumReads</th>\n",
       "      <th>ratioPercent</th>\n",
       "      <th>ratioLog</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Day_0_Xi_Mus_1</td>\n",
       "      <td>0610009B22Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Day_0_Xi_Mus_1</td>\n",
       "      <td>0610009O20Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Day_0_Xi_Mus_1</td>\n",
       "      <td>0610010F05Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Day_0_Xi_Mus_1</td>\n",
       "      <td>0610010K14Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Day_0_Xi_Mus_1</td>\n",
       "      <td>0610030E20Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "genome            Name           gene  129  Cast  sumReads  ratioPercent  \\\n",
       "0       Day_0_Xi_Mus_1  0610009B22Rik    0     0         0           NaN   \n",
       "1       Day_0_Xi_Mus_1  0610009O20Rik    0     0         0           NaN   \n",
       "2       Day_0_Xi_Mus_1  0610010F05Rik    0     0         0           NaN   \n",
       "3       Day_0_Xi_Mus_1  0610010K14Rik    0     0         0           NaN   \n",
       "4       Day_0_Xi_Mus_1  0610030E20Rik    0     0         0           NaN   \n",
       "\n",
       "genome  ratioLog  \n",
       "0            NaN  \n",
       "1            NaN  \n",
       "2            NaN  \n",
       "3            NaN  \n",
       "4            NaN  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=matrix2ratio(df)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "df.to_csv(\"/staging/leuven/stg_00041/Adrian/TALON_JANISZEWSKI_XCR2/allele_specific/pre_processing/AJ_0065.9_allelic_bart_zeros_included.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:allele_specific]",
   "language": "python",
   "name": "conda-env-allele_specific-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
